### PR TITLE
Escaping JSON strings in QuarkusErrorServlet

### LINKED ISSUE #9463: Bad JSON returned from the default QuarkusErrorServlet
**Describe the bug**
Using quarkus-undertow and quarkus-resteasy, if an exception is thrown in a JAX-RS endpoint and no ExceptionMapper catches it gets forwarded to QuarkusErrorServlet. 

In case of JSON Accept header the stack does not get properly escaped there.

For example, I've encountered a stack ending with (there's a tab before ....):
`(SynchronousDispatcher.java:488)\n	... 53 more`
which produces a JSON with a string including a tab delimiter.

Here is an example of a more thorough escaping from JSON simple.
https://github.com/fangyidong/json-simple/blob/master/src/main/java/org/json/simple/JSONValue.java#L270

**Environment:**
 - Quarkus version 1.4.2.Final

### PR DESCRIPTION
**Pull request by lodrantl**
Fixes #9463.

Just added a more thorough escaping for the strings.

### CODE REVIEW

File: extensions/undertow/runtime/src/main/java/io/quarkus/undertow/runtime/QuarkusErrorServlet.java [hunk 1]
```diff
@@ -40,15 +40,17 @@ protected void service(HttpServletRequest req, HttpServletResponse resp) throws
...
-            StringBuilder jsonPayload = new StringBuilder("{\"details\":\"").append(details).append("\",\"stack\":\"")
-                    .append(escapedStack).append("\"}");
+            String escapedDetails = escapeJsonString(details);
+            String escapedStack = escapeJsonString(stack);
+            StringBuilder jsonPayload = new StringBuilder("{\"details\":\"").append(escapedDetails)
+                    .append("\",\"stack\":\"").append(escapedStack).append("\"}");
             resp.getWriter().write(jsonPayload.toString());
         } else {
             //We default to HTML representation
+            String escapedDetails = escapeHtml(details);
```
Comment (gsmet) [inline extensions/undertow/runtime/src/main/java/io/quarkus/undertow/runtime/QuarkusErrorServlet.java:50]: I think it is escaped later in `TemplateHtmlBuilder` so we don't need to escape it here. Am I right?

**Review by gsmet — CHANGES_REQUESTED** [review]
Thanks! I added a small question. I think the details are currently escaped twice when using HTML but that would need to be properly checked.

File: extensions/undertow/runtime/src/main/java/io/quarkus/undertow/runtime/QuarkusErrorServlet.java (hunk 1 above)
Comment (lodrantl) [reply extensions/undertow/runtime/src/main/java/io/quarkus/undertow/runtime/QuarkusErrorServlet.java]: Yes, it is also escaped in the `TemplateHtmlBuilder` directly in the used constructor. I just removed the escaping from QuarkusErrorServlet.

**Comment by geoand** [comment]
Could you please rebase onto master? There are some CI fixes in there that should allow the PR to pass

**Review by gsmet — APPROVED** [review]

**Comment by gsmet** [comment]
Merged, thanks!

**Pull request merged**